src/Consumer/Process.php

```diff
@@ -644,21 +644,22 @@ public function succFetch(array $result, int $fd): void
                     continue;
                 }
 
-                $offset = $assign->getConsumerOffset($topic['topicName'], $part['partition']);
-
-                if ($offset === null) {
+                $consumerOffset = $assign->getConsumerOffset($topic['topicName'], $part['partition']);
+                if ($consumerOffset === null) {
                     return; // current is rejoin....
                 }
 
+                $commitOffset = $consumerOffset - 1;
                 foreach ($part['messages'] as $message) {
                     $this->messages[$topic['topicName']][$part['partition']][] = $message;
 
-                    $offset = $message['offset'];
+                    $commitOffset = $message['offset'];
                 }
 
-                $consumerOffset = ($part['highwaterMarkOffset'] > $offset) ? ($offset + 1) : $offset;
+                $consumerOffset = $commitOffset + 1;
```

Comment:
here shouldn't + 1

Reply:
consumer offset is equal to commit offset + 1, so I add 1 to the commit offset, and consumer will fetch the next message. If you don't add 1 to commit offset, you will fetch the same message committed last time.

Reply:
you said right but the pos is wrong ref #189

Reply:
i use your code but i still lost message, #186  is the condition i meet，but the test case i could not counstruct...

Reply:
@noname007 can you post the kafka-php log as a attachment? I test my code many times, but it doesn't occur!

Reply:
@noname007 This is my code:
public function succFetch($result, $fd)
    {
        $assign = \Kafka\Consumer\Assignment::getInstance();
        $this->debug('Fetch success, result:' . json_encode($result));
        foreach ($result['topics'] as $topic) {
            foreach ($topic['partitions'] as $part) {
                $context = array(
                    $topic['topicName'],
                    $part['partition'],
                );
                if ($part['errorCode'] != 0) {
                    $this->stateConvert($part['errorCode'], $context);
                    continue;
                }

                $offset = $assign->getConsumerOffset($topic['topicName'], $part['partition']);
                if ($offset === false) {
                    return; // current is rejoin....
                }

                $commitOffset = $consumerOffset - 1;
                foreach ($part['messages'] as $message) {
                    $this->messages[$topic['topicName']][$part['partition']][] = $message;

                    $commitOffset = $message['offset'];
                }

                $consumerOffset = $commitOffset + 1;

                $assign->setConsumerOffset($topic['topicName'], $part['partition'], $consumerOffset);
                $assign->setCommitOffset($topic['topicName'], $part['partition'], $commitOffset);
            }
        }
        $this->state->succRun(\Kafka\Consumer\State::REQUEST_FETCH, $fd);
    }

Reply:
你这个和我那里的下面的等价的，我的是上边的那个地方多加1，导致的。

Reply:
@chongchaoyu https://github.com/weiboad/kafka-php/pull/186/files#diff-abf40442ee37f59d8941040a616cf543R121 这个就是浮现问题的脚本，但是放到测试里面，没有按照我的本地脚本的方式预期的跑。直接抛异常，然后再消费，数据就丢失了一条。

Reply:
@noname007 I understand that English is not your native language (it's also not mine), but I'd really love if I don't have to translate comments in order to help people...

Reply:
i said nothing.....just say the script run mode  in my machine that could cause the problem:

throw a exception after it run ,and run again ,and will find the offset wrong in the log file .

but the test case i write was not run as i thought because i could write the write assert statement.

Reply:
> because i found he is a Chinese man in the QQ group so i select our native lang to talk... sorry for that.

No worries, it happens 😄

Reply:
@lcobucci here is my run scripts coditions. i use the  crontable to check the process job active or not ,if it is not then pull up by crobtable